### PR TITLE
add getGlobalConstraintParameters to Controller and UController

### PR DESCRIPTION
**Pull request by dkent600**
resolves: #470 

- adds `Controller.getGlobalConstraintParameters` to Controller and UController
- renames "register" to "isRegistered" in `GlobalConstraintRegister `
- raises gasLimit in package.json and constants.js so tests will succeed

### CODE REVIEW

File: contracts/controller/Controller.sol [hunk 1]
```diff
@@ -257,6 +257,20 @@ contract Controller is ControllerInterface {
         return schemes[_scheme].permissions;
     }
 
+    function getGlobalConstraintParameters(address _globalConstraint,address) external view returns(bytes32) {
```
Comment (orenyodfat) [inline contracts/controller/Controller.sol]: Please return here 2 values  : pre and post params  .
As gc can have both.

File: contracts/controller/UController.sol [hunk 2]
```diff
@@ -280,6 +280,23 @@ contract UController is ControllerInterface {
         return organizations[_avatar].schemes[_scheme].permissions;
     }
 
+    function getGlobalConstraintParameters(address _globalConstraint, address _avatar) external view returns(bytes32) {
```
Comment (orenyodfat) [inline contracts/controller/UController.sol]: please return 2 value : pre and post params

File: package.json [hunk 3]
```diff
@@ -11,9 +11,6 @@
     "truffle.js",
     "tsconfig.json"
   ],
-  "config": {
-    "gasLimit": "5600000"
```
Comment (orenyodfat) [inline package.json]: Why is this removed ?
pr #472 need that,

File: contracts/controller/Controller.sol (hunk 1 above)
Comment (dkent600) [reply contracts/controller/Controller.sol]: As I read the code, both pre and post will have exactly the same params.  Is that not the case?

File: contracts/controller/UController.sol (hunk 2 above)
Comment (dkent600) [reply contracts/controller/UController.sol]: As I read the code, both pre and post will have exactly the same params. Is that not the case?

File: truffle.js [hunk 4]
```diff
@@ -10,31 +10,31 @@ module.exports = {
       network_id: 1,
       host: "localhost",
       port: 8546,
-      gas: 4543760
+      gas: 4600000
```
[the remark this answers is not in the document]
Comment (dkent600) [reply truffle.js]: I can't explain why that change was needed at the time.  Now it isn't, so I"ve restored it.

File: package.json (hunk 3 above)
Comment (dkent600) [reply package.json]: restored

File: truffle.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (orenyodfat) [reply truffle.js]: on which call it is failing ?

[the remark this answers is not in the document]
Comment (dkent600) [reply truffle.js]: >>>
Running migration: 1_initial_migration.js
  Deploying Migrations...
  ... 0xe3d6943c21fe39152016238fbd67f80b94f27b5670ef994fef19f5e86081c2bc
  Migrations: 0xc3415220f7dedf287379a536abd275d837991b3a
Saving successful migration to network...
  ... 0xd64b06965cadfb11137eb0e9e2e7102f761e14e5b5fd81719b9757907753c699
Saving artifacts...
Running migration: 2_deploy_organization.js
  Deploying ControllerCreator...
  ... 0xff5c9f2cff8ee3ba0f6522ca21f45cd1f573578b5e8b08f003e94e382b00d787
Error encountered, bailing. Network state unknown. Review successful transactions manually.
Error: VM Exception while processing transaction: out of gas
>>>

[the remark this answers is not in the document]
Comment (orenyodfat) [reply truffle.js]: migration pass without this modification.
please recheck.

[the remark this answers is not in the document]
Comment (dkent600) [reply truffle.js]: Did you try it with _this_ branch?  I already did the recheck after your query above.  All I had to do was revert the gas limit and the travis build fails.

File: contracts/controller/UController.sol [hunk 5]
```diff
@@ -280,6 +280,23 @@ contract UController is ControllerInterface {
...
+
+        Organization organization = organizations[_avatar];
```
[the remark this answers is not in the document]
Comment (dkent600) [reply contracts/controller/UController.sol]: added `storage`

File: contracts/controller/Controller.sol [hunk 6]
```diff
@@ -257,6 +257,20 @@ contract Controller is ControllerInterface {
...
+        GlobalConstraintRegister register = globalConstraintsRegisterPre[_globalConstraint];
```
[the remark this answers is not in the document]
Comment (dkent600) [reply contracts/controller/Controller.sol]: added `memory`

File: contracts/controller/UController.sol [hunk 7]
```diff
@@ -280,6 +280,23 @@ contract UController is ControllerInterface {
...
+
+        GlobalConstraintRegister register = organization.globalConstraintsRegisterPre[_globalConstraint];
```
[the remark this answers is not in the document]
Comment (dkent600) [reply contracts/controller/UController.sol]: added `memory`

File: truffle.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (orenyodfat) [reply truffle.js]: Yes.this branch .
you should set the gasLimit though the change in truffle.js is not needed.

[the remark this answers is not in the document]
Comment (dkent600) [reply truffle.js]: Update:  After [adding](https://github.com/daostack/arc/pull/471/commits/dd17075ed59910ba4cf2cbd5c87051f3baff4111) the `storage` and `memory` qualifiers, the gas requirement apparently increased, causing the build to fail again.  Tests also started failing, requiring anther raise in the constants.js gasLimit.

Here is the build output for the failing migration:

https://travis-ci.org/daostack/arc/builds/390725020?utm_source=github_status&utm_medium=notification

Thus I have further increased the limit in truffle.js.

[the remark this answers is not in the document]
Comment (dkent600) [reply truffle.js]: I reverted this change and replaced it by manually supplying the `constants.GENESIS_SCHEME_GAS_LIMIT` to the deployment of `ControllerCreator` during migration and to all calls to `ControllerCreator.new` in the tests.
